pkg/relayer/session/sessiontree_test.go

```diff
@@ -1,3 +1,96 @@
 package session_test
 
-// TODO: Add tests to the sessionTree logic
+import (
+	"bytes"
+	"compress/gzip"
+	"crypto/rand"
+	"testing"
+
+	"github.com/pokt-network/poktroll/pkg/crypto/protocol"
+	"github.com/pokt-network/smt"
+	"github.com/pokt-network/smt/kvstore/pebble"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSessionTree_CompactProofsAreSmallerThanNonCompactProofs(t *testing.T) {
+	// Run the test for different number of leaves.
+	for numLeaf := 10; numLeaf <= 1000000; numLeaf *= 10 {
+		// We run the test 1000 times for each number of leaves to remove the randomness bias.
+		cumulativeProofSize := 0
+		cumulativeCompactProofSize := 0
+		cumulativeGzippedProofSize := 0
+		for numLeaf := 0; numLeaf <= 1000; numLeaf++ {
+			kvStore, err := pebble.NewKVStore("")
+			require.NoError(t, err)
+
+			trie := smt.NewSparseMerkleSumTrie(kvStore, protocol.NewTrieHasher(), smt.WithValueHasher(nil))
+
+			// Insert numLeaf random leaves.
+			for i := 0; i < numLeaf; i++ {
+				key := make([]byte, 32)
+				_, err := rand.Read(key)
+				require.NoError(t, err)
+				// Insert an empty value since this does not get affected by the compaction,
+				// this is also to not favor proof compression that compresses the value too.
+				trie.Update(key, []byte{}, 1)
+			}
+
+			// Generate a random path.
+			var path = make([]byte, 32)
+			_, err = rand.Read(path)
+			require.NoError(t, err)
+
+			// Create the proof.
+			proof, err := trie.ProveClosest(path)
+			require.NoError(t, err)
+
+			proofBz, err := proof.Marshal()
+			require.NoError(t, err)
+
+			// Accumulate the proof size over 1000 runs.
```

Comment:
Don't put numbers like this in the comments.

If you (or another developer) ever update the loop, there's no chance you'll update the comment.

Comments should be "ever green".

If they're not "ever green", they should involve a TODO.